torch/utils/tensorboard/proto_graph.py

```diff
@@ -0,0 +1,39 @@
+from tensorboard.compat.proto.node_def_pb2 import NodeDef
+from tensorboard.compat.proto.attr_value_pb2 import AttrValue
+from tensorboard.compat.proto.tensor_shape_pb2 import TensorShapeProto
+
+
+def AttrValue_proto(dtype, shape, s):
+    attr = {}
+
+    if s is not None:
+        attr['attr'] = AttrValue(s=s.encode(encoding='utf_8'))
```

Comment:
Does `'attr'` correspond to some attribute convention in Tensorboard? If so, can we reference the relevant reference documentation from this PR?

Reply:
@lanpa can you comment on this one? Thanks.

Reply:
This is derived from reverse engineering, I have to find the reference.

Reply:
Adding proto references.